apps/ejabberd/test/ejabberd_listener_SUITE.erl

```diff
@@ -0,0 +1,31 @@
+-module(ejabberd_listener_SUITE).
+
+-compile([export_all]).
+
+all() ->
+    [tcp_socket_is_started_with_options].
+
+
+tcp_socket_is_started_with_options(_C) ->
+
+    meck:new(gen_tcp, [unstick, passthrough]),
+    meck:expect(gen_tcp, listen, fun(Port, Opts) ->
+                                         meck:passthrough([Port, Opts])
+                                 end),
+
+    OverrideBacklog = {backlog, 50},
+    {ok, _Pid} = listener_started([OverrideBacklog]),
+
+    [{_Pid, {gen_tcp, listen, [_, Opts]}, _Result}] =  meck:history(gen_tcp),
+
+    DefaultBacklog = {backlog, 100},
+    [DefaultBacklog, OverrideBacklog] = proplists:lookup_all(backlog, Opts),
```

Comment:
See this https://github.com/esl/MongooseIM/pull/708#discussion_r56475918 for explanation.